modules/account-lib/src/coin/hbar/transaction.ts

```diff
@@ -26,189 +25,117 @@ export class Transaction extends BaseTransaction {
+  innerTransaction(tx: SDKTransaction) {
     this._hederaTx = tx;
-    // this.loadPreviousSignatures();
-  }
-
-  /**
-   * Decode previous signatures from the inner hedera transaction
-   * and save them into the base transaction signature list.
-   */
-  loadPreviousSignatures(): void {
-    if (this._hederaTx.sigMap && this._hederaTx.sigMap.sigPair) {
-      const sigPairs = this._hederaTx.sigMap.sigPair;
-      sigPairs.forEach(sigPair => {
-        const signature = sigPair.ed25519;
-        if (signature) {
-          this._signatures.push(toHex(signature));
-        }
-      });
-    }
-  }
-
-  /**
-   * Sets this transaction body components
-   *
-   * @param {Uint8Array} bytes encoded body transaction
-   */
-  bodyBytes(bytes: Uint8Array) {
-    this.body(proto.Transaction.decode(bytes));
-  }
-  //endregion
-
-  //region helpers
-  /**
-   * Returns this hedera transaction id components in a readable format
-   *
-   * @returns {[string, string]} - transaction id parts [<account id>, <startTime in seconds>]
-   */
-  getTxIdParts(): [string, string] {
-    if (
-      this._txBody &&
-      this._txBody.transactionID &&
-      this._txBody.transactionID.accountID &&
-      this._txBody.transactionID.transactionValidStart
-    ) {
-      return [
-        stringifyAccountId(this._txBody.transactionID.accountID),
-        stringifyTxTime(this._txBody.transactionID.transactionValidStart),
-      ];
-    }
-    throw new Error('Missing transaction id information');
-  }
-
-  /**
-   * Returns this transaction hash
-   *
-   * @returns {string} - The transaction hash
-   */
-  getTxHash(): string {
-    if (!this._txBody.nodeAccountID) {
-      throw new Error('Missing transaction node id');
-    }
-    return this.getHashOf(this._hederaTx);
-  }
-
-  /**
-   * Encode an object using the given encoder class
-   *
-   * @param obj - the object to be encoded, it must be an proto namespace object
-   * @param encoder - Object encoder
-   * @returns {Uint8Array} - encoded object byte array
-   */
-  private encode<T extends { constructor: Function }>(obj: T, encoder?: { encode(arg: T): Writer }): Uint8Array {
-    if (encoder) {
-      return encoder.encode(obj).finish();
-    }
-    return this.encode(obj, proto[obj.constructor.name]);
-  }
-
-  /**
-   * Returns an sha-384 hash
-   *
-   * @param {Uint8Array} bytes - bytes to be hashed
-   * @returns {string} - the resulting hash string
-   */
-  sha(bytes: Uint8Array): string {
-    return toHex(hash(bytes));
   }
 
   /**
-   * Returns a hash of the given proto object.
+   * Set the inner SDK transaction as bytes
    *
-   * @param obj - The object to be hashed, it must be an proto namespace object
-   * @returns {string} - the resulting hash string
+   * @param {Uint8Array} bytes encoded SDK transaction
    */
-  private getHashOf<T>(obj: T): string {
-    return this.sha(this.encode(obj));
+  fromBytes(bytes: Uint8Array) {
```

Comment:
Shouldn't this be a `constructor`? I imagine we use this when we create it a transaction.

Reply:
Yes that actually sounds really good we'll try to add an alternative constructor